### ISSUE TITLE
Not support login by 'Continue With Google'

### ISSUE THREAD

**Issue by StarryTang** [LiLittleCat/intellij-chatgpt#7]
![image](https://user-images.githubusercontent.com/108268264/209047297-f3c35ead-88cf-4974-8d2b-b9a535691b38.png)

**Comment by StarryTang** [comment]
When I log in through 'Continue With Google',it ask that the current browser is not secure and I cannot log in.

**Comment by KrunchMuffin** [comment]
same

Couldn’t sign you in
This browser or app may not be secure.
Try using a different browser. If you’re already using a supported browser, you can try again to sign in.

**Comment by LiLittleCat** [comment]
@StarryTang @KrunchMuffin Sorry for the late reply, it seems that Google does not think JCEF is safe, I will ask the JetBrains team to see how to solve this.

**Comment by LiLittleCat** [comment]
@emahuni, Of course, you can, but the token you get through https://chat.openai.com/api/auth/session is an access token for a session. The token needed to enter in settings is a session token, you can get a session token by the steps:
1. Go to https://chat.openai.com/chat and log in or sign up.
2. Open dev tools.
3. Open `Application` > `Cookies`.
  [![pSSKdmR.png](https://s1.ax1x.com/2022/12/28/pSSKdmR.png)](https://imgse.com/i/pSSKdmR)
4. Copy the value for `__Secure-next-auth.session-token` as settings value.
  [![pSSK6pD.png](https://s1.ax1x.com/2022/12/28/pSSK6pD.png)](https://imgse.com/i/pSSK6pD)
After you enter the session token, you need to restart the IDE because ChatGPT Tool Windows needs a restart, and you won't need to log in until the session token is expired.

**Comment by dozer111** [comment]
> @emahuni, Of course, you can, but the token you get through https://chat.openai.com/api/auth/session is an access token for a session. The token needed to enter in settings is a session token, you can get a session token by the steps:
> 
> 1. Go to https://chat.openai.com/chat and log in or sign up.
> 2. Open dev tools.
> 3. Open `Application` > `Cookies`.
>    [![pSSKdmR.png](https://camo.githubusercontent.com/6d0ade8eb499066e0f346f3feedc287ed572ab6163e88a561de6adec0a60871a/68747470733a2f2f73312e617831782e636f6d2f323032322f31322f32382f7053534b646d522e706e67)](https://imgse.com/i/pSSKdmR)
> 4. Copy the value for `__Secure-next-auth.session-token` as settings value.
>    [![pSSK6pD.png](https://camo.githubusercontent.com/466f56bd880a050ff63a0d4f94fff18fa7feae79a40ac942694bb62e968c32bb/68747470733a2f2f73312e617831782e636f6d2f323032322f31322f32382f7053534b3670442e706e67)](https://imgse.com/i/pSSK6pD)
> 
> After you enter the session token, you need to restart the IDE because ChatGPT Tool Windows needs a restart, and you won't need to log in until the session token is expired.

sorry for stupid question, but I have(or don't see) this field. Where is it?(or maybe from what version is it appears)?

**Comment by emahuni** [comment]
> @emahuni, Of course, you can, but the token you get through https://chat.openai.com/api/auth/session is an access token for a session. The token needed to enter in settings is a session token, you can get a session token by the steps:
> 
> 1. Go to https://chat.openai.com/chat and log in or sign up.
> 2. Open dev tools.
> 3. Open `Application` > `Cookies`.
>    [![pSSKdmR.png](https://camo.githubusercontent.com/6d0ade8eb499066e0f346f3feedc287ed572ab6163e88a561de6adec0a60871a/68747470733a2f2f73312e617831782e636f6d2f323032322f31322f32382f7053534b646d522e706e67)](https://imgse.com/i/pSSKdmR)
> 4. Copy the value for `__Secure-next-auth.session-token` as settings value.
>    [![pSSK6pD.png](https://camo.githubusercontent.com/466f56bd880a050ff63a0d4f94fff18fa7feae79a40ac942694bb62e968c32bb/68747470733a2f2f73312e617831782e636f6d2f323032322f31322f32382f7053534b3670442e706e67)](https://imgse.com/i/pSSK6pD)
> 
> After you enter the session token, you need to restart the IDE because ChatGPT Tool Windows needs a restart, and you won't need to log in until the session token is expired.

Sweet! it works

**Comment by emahuni** [comment]
I updated Webstorm to the latest and it updated your plugin, now there is nowhere to enter that token anymore, just url something something. what happened?

**Comment by LiLittleCat** [comment]
> I updated Webstorm to the latest and it updated your plugin, now there is nowhere to enter that token anymore, just url something something. what happened?

Hi @emahuni, I wanted to let you know that I removed the token settings because it was causing an issue, but unfortunately, I wasn't able to find a solution for it. You can see more details about the problem in this GitHub issue: https://github.com/LiLittleCat/intellij-chatgpt/issues/16#issuecomment-1477877290.

I was wondering, are you using a Google account to log in to the official website? If the token settings are necessary, perhaps I should bring them back and work on finding a solution to the issue.

**Comment by jmigliavacca** [comment]
Yes, it's needed. I cannot log in otherwise. If I use google, it won't let me in because of browser security concerns, and openai doesn't let me change the authentication method.

**Comment by emahuni** [comment]
> Yes, it's needed. I cannot log in otherwise. If I use google, it won't let me in because of browser security concerns, and openai doesn't let me change the authentication method.

Exactly

**Comment by skorpions2000** [comment]
I noticed your dialog, don't know if it's related?... I described my problem here... https://github.com/LiLittleCat/intellij-chatgpt/issues/28 Unfortunately the plugin doesn't work at all...
